src/averageAges.js

```diff
@@ -1,61 +1,55 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mens = findSex(people, 'm');
+
+  const withCentury = century > 0
+    ? mens.filter(men => Math.ceil(men.died / 100) === century)
+    : mens;
+
+  return findAverageAge(withCentury);
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womens = findSex(people, 'f');
+
+  const haveChildren = withChildren
+    ? womens.filter(women =>
+      people.find(person => women.name === person.mother))
+    : womens;
+
+  return findAverageAge(haveChildren);
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => {
+    return people.find(person => person.name === child.mother);
+  });
+
+  const isSon = onlyWithSon
+    ? findSex(children, 'm')
+    : children;
+
+  const sumDifference = isSon.reduce((sum, son) => {
+    return sum + son.born
+      - people.find(person => person.name === son.mother).born;
+  }, 0);
+
+  return sumDifference / isSon.length;
+}
+
+function findSex(people, sex) {
+  const peoples = people.filter(person => person.sex === sex);
+
+  return peoples;
+}
```

Comment:
Use proper names for variables. If function was called 'makeFruit', it should return 'fruit'.